src/index.rs

```diff
@@ -984,4 +1003,198 @@ mod tests {
         assert!(new2 > old2);
         assert!(new2 > new1);
     }
+
+    #[test]
+    fn should_correctly_update_one_resource() {
```

Comment:
This test case is not correct. See test case `fn should_not_update_nonexistent_path`.